admin_panel/src/commonMain/kotlin/net/thechance/mena/admin_panel/domain/repository/user/UserRepository.kt

```diff
@@ -0,0 +1,15 @@
+package net.thechance.mena.admin_panel.domain.repository.user
+
+
+import net.thechance.mena.admin_panel.domain.entity.user.Status
+import net.thechance.mena.admin_panel.domain.entity.user.User
+import net.thechance.mena.admin_panel.domain.model.UserQuery
+import kotlin.uuid.ExperimentalUuidApi
+import kotlin.uuid.Uuid
+@OptIn(ExperimentalUuidApi::class)
+interface UserRepository {
+    suspend fun getUsers(userQuery: UserQuery?):List<User>
+
+    suspend fun updateUserStatus(userID: Uuid , status : Status)
+
+}
```

Comment:
i think this better?

```suggestion

import net.thechance.mena.admin_panel.domain.entity.user.Status
import net.thechance.mena.admin_panel.domain.entity.user.User
import net.thechance.mena.admin_panel.domain.model.UserQuery
import kotlin.uuid.ExperimentalUuidApi
import kotlin.uuid.Uuid

@OptIn(ExperimentalUuidApi::class)
interface UserRepository {
    suspend fun getUsers(userQuery: UserQuery?): List<User>
    suspend fun updateUserStatus(userID: Uuid, status : Status)
}
```